presto-main/src/main/java/com/facebook/presto/operator/scalar/IpPrefixFunctions.java

```diff
@@ -71,4 +74,78 @@ public static Slice stringIpPrefix(@SqlType("varchar(x)") Slice slice, @SqlType(
     {
         return ipPrefix(castFromVarcharToIpAddress(slice), subnetSize);
     }
+
+    @Description("Smallest IP address for a given IP prefix")
+    @ScalarFunction("ip_subnet_min")
+    @SqlType(StandardTypes.IPADDRESS)
+    public static Slice ipSubnetMin(@SqlType(StandardTypes.IPPREFIX) Slice value)
+    {
+        return castFromIpPrefixToIpAddress(value);
+    }
+
+    @Description("Largest IP address for a given IP prefix")
+    @ScalarFunction("ip_subnet_max")
+    @SqlType(StandardTypes.IPADDRESS)
+    public static Slice ipSubnetMax(@SqlType(StandardTypes.IPPREFIX) Slice value)
+    {
+        byte[] address = toInetAddress(value.slice(0, IPADDRESS.getFixedSize())).getAddress();
+        int subnetSize = value.getByte(IPPREFIX.getFixedSize() - 1) & 0xFF;
+
+        if (address.length == 4) {
+            for (int i = 0; i < 4; i++) {
+                address[3 - i] |= (byte) ~(~0 << min(max((32 - subnetSize) - 8 * i, 0), 8));
+            }
+            byte[] bytes = new byte[16];
+            bytes[10] = (byte) 0xff;
+            bytes[11] = (byte) 0xff;
+            arraycopy(address, 0, bytes, 12, 4);
+            address = bytes;
+        }
+        else if (address.length == 16) {
+            for (int i = 0; i < 16; i++) {
+                address[15 - i] |= (byte) ~(~0 << min(max((128 - subnetSize) - 8 * i, 0), 8));
+            }
+        }
+        return wrappedBuffer(address);
+    }
+
+    @Description("Array of smallest and largest IP address in the subnet of the given IP prefix")
+    @ScalarFunction("ip_subnet_range")
+    @SqlType("array(IPADDRESS)")
+    public static Block ipSubnetRange(@SqlType(StandardTypes.IPPREFIX) Slice value)
+    {
+        BlockBuilder blockBuilder = IPADDRESS.createBlockBuilder(null, 2);
+        IPADDRESS.writeSlice(blockBuilder, ipSubnetMin(value));
+        IPADDRESS.writeSlice(blockBuilder, ipSubnetMax(value));
+        return blockBuilder.build();
+    }
+
+    @Description("Is the IP address in the subnet of IP prefix")
+    @ScalarFunction("is_subnet_of")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean isSubnetOf(@SqlType(StandardTypes.IPPREFIX) Slice left, @SqlType(StandardTypes.IPADDRESS) Slice right)
+    {
+        toInetAddress(right);
+        return between(right, ipSubnetMin(left), ipSubnetMax(left));
+    }
+
+    @Description("Is the second IP prefix argument within the subnet of the first IP prefix argument")
+    @ScalarFunction("is_subnet_of")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean isPrefixSubnetOf(@SqlType(StandardTypes.IPPREFIX) Slice left, @SqlType(StandardTypes.IPPREFIX) Slice right)
```

Comment:
nit: What about naming those variables as `first` and `second`, as corresponding to the description.